src/js/preshuffle.s

```diff
@@ -809,55 +802,81 @@ MaybeSetCheckpointActual:
   .byte $3a,$2b
 .org $8ed3 ; Experience
   .byte $5a,$2b
-; Experience to next level is now unused,
-; if you want to use it for something, make sure the drawing isn't skipped
-; .org $8ed9 ; Experience Cap
-;   .byte $5a,$2b
+
+;;; Numeric displays
+.org $8ed7  ; 03 - was Exp Left, now its Max MP
+  .word (PlayerMaxMP)
+  .byte $7b,$2b,$02,$00 ; copied from $34ee3
+.org $8ee3  ; 05 - was Max MP, now its unused
+.org $8ee9  ; 06 - was LV(menu) but now it's difficulty
+  .word (Difficulty)
+  .byte $56,$2b,$03,$00 ; display left of exp
+.org $8f19  ; 0e - was unused, now it's LV(menu)
+  .word (PlayerLevel)
+  .byte $29,$29,$03,$00 ; copied from $34ee9
 
 .pushseg "13", "fe", "ff"
-;; Prevent EXP Left from being drawn
 ;InitializeStatusBarNametable
-.org $bace
-  jsr DisplayNumberInternalButSkipExpToNextLevel
-.reloc
-DisplayNumberInternalButSkipExpToNextLevel:
-  cmp #$03
-  beq +
-  jmp DisplayNumberInternal
-+ rts
+.org $baca
+  lda #$5f ; 0b0101'1111 - update all 5 status display (including difficulty)
+  jsr UpdateStatusBarDisplays
+  jmp $c676 ; WaitForNametableFlush
+FREE_UNTIL $bad9
+.popseg
+
+.pushseg "1a", "fe", "ff"
+;;; Calls DisplayNumberInternal for each of the bits in A as follows:
+;; [in] A - bit mask for the different displays to update
+;; [scratch] X - could push X if needed later
+;; The following values are used in the randomizer for this
+;; 0 - Level
+;; 1 - Money
+;; 2 - Exp
+;; 3 - Max MP
+;; 4 - MP
+;; 5 - (currently unused)
+;; 6 - Difficulty
+.reloc
+UpdateStatusBarDisplays:
+  ldx #$07
+- rol
```

Comment:
(nit) would you mind indenting the lines between `- rol` (inclusive) and `bpl -` (exclusive) to more clearly demarcate the loop?